models/users.js

```diff
@@ -163,53 +163,74 @@ const getSuggestedUsers = async (skill) => {
  * @return {Promise<userModel|Array>}
  */
 const fetchPaginatedUsers = async (query) => {
+  const isDevMode = query.dev === "true";
+
   try {
-    // INFO: default user size set to 100
-    // INFO: https://github.com/Real-Dev-Squad/website-backend/pull/873#discussion_r1064229932
     const size = parseInt(query.size) || 100;
     const doc = (query.next || query.prev) && (await userModel.doc(query.next || query.prev).get());
 
     let dbQuery = userModel.where("roles.archived", "==", false).orderBy("username");
+    let compositeQuery = [dbQuery];
+    if (isDevMode) {
+      const usernameQuery = userModel.where("roles.archived", "==", false).orderBy("username");
+      const firstNameQuery = userModel.where("roles.archived", "==", false).orderBy("first_name");
+      const lastNameQuery = userModel.where("roles.archived", "==", false).orderBy("last_name");
+      compositeQuery = [usernameQuery, firstNameQuery, lastNameQuery];
+    }
 
     if (query.prev) {
+      compositeQuery = compositeQuery.map((query) => query.limitToLast(size));
       dbQuery = dbQuery.limitToLast(size);
     } else {
+      compositeQuery = compositeQuery.map((query) => query.limit(size));
       dbQuery = dbQuery.limit(size);
```

Comment:
As dbQuery will itself lie inside composite query, why do we need to use it here as well?